build-release-tools/lib/manifest.py

```diff
@@ -324,6 +323,97 @@ def check_commit_changed(repo, repo_url, branch, commit):
         return False
 
     @staticmethod
+    def check_branch_changed(repo, repo_url, branch, commit):
+        """
+        Check whether the repository is changed based on its url, branch ,commit-id and arguments repo_url, branch, commit
+        :param repo: an repository entry of member _repositories or _downstream_jobs
+        :param repo_url: the url of the repository
+        :param branch: the branch of the repository
+        :param commit: the commit id of the repository
+        :return: True when the branch is different with argument branch
+                 and the url and commit is the same with the arguments repo_url, commit;
+                 otherwise, False
+        """
+
+        if (repo['repository'] == repo_url):
+            # If repo has "branch", compare "commit-id" in repo with the argument commit
+            # only when "branch" is the same with argument branch.
+
+            if 'commit-id' in repo:
+                if repo['commit-id'] != commit:
+                    return False
+            
+            # Exits with 1 if repo doesn't have "branch" and "commit-id"
+            elif 'branch' not in repo:
+                    raise KeyError("Neither commit-id nor branch is set for repository {0}".format(repo['repository']))
```

Comment:
user of this class should handle with the exception.
or just end their running scripts when exception occurs.